common/src/main/java/com/microsoft/identity/common/internal/util/RestrictionsManagerHelper.kt

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.util
+
+import android.content.Context
+import android.content.RestrictionsManager
+import android.os.Bundle
+import com.microsoft.identity.common.internal.broker.ipc.BrokerProcessIpcUtils
+import com.microsoft.identity.common.components.AndroidPlatformComponentsFactory
+
+import com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle
+import com.microsoft.identity.common.internal.broker.PackageHelper
+import com.microsoft.identity.common.java.interfaces.IPlatformComponents
+import com.microsoft.identity.common.logging.Logger
+
+/**
+ * Helper class to read the app restrictions manager of the current broker app or another broker app.
+ */
+class RestrictionsManagerHelper(
```

Comment:
>RestrictionsManagerHelper [](http://example.com/codeflow?start=6&length=25)

If you're going to use this in WebViewAuthorizationFragment, how will it work in OneAuth only flow?

Reply:
If we are in a non-broker flow, we do not call RestrictionsManagerHelper for this QR PIN scenario.

However, in a hypothetical scenario where RestrictionsManagerHelper is used in a non-broker flow:
1. Requesting data from a broker app: will make an IPC call to the broker app, read the Restrictions manager from there, and return the result.
2. Requesting data from the current app: will read the Restrictions manager of the app, which will likely be null.

Reply:
I raised this because I feel like this is broker only component is being put in common and can potentially run in non-brokered flow. It would be better if could inject an object in common layer (or WebViewAuthorizationFragment in this case) to _customize_ behavior for broker vs other. It's general problem that is seen on other cases as well e.g. nonce redirect handling, flighting 

I guess, it might be too much work if you do that in this PR?

Reply:
Q: For what would be logic for non-broker flow?

Reply:
for non-broker flow we do not check the restriction manager, and we just grant the camera permission if we have it